src/actions/actions.js

```diff
@@ -302,21 +346,10 @@ export function saveNotebookToServer() {
             details: `${message} <br />Notebook saved`,
           }))
         })
+      dispatch({ type: 'NOTEBOOK_SAVED' })
     } else {
-      // Create a New Notebook in Database
-      fetch('/api/v1/notebooks/', postRequestOptions)
-        .then(response => response.json())
-        .then((json) => {
-          const message = 'Notebook saved to server'
-          dispatch(updateAppMessages({
-            message,
-            details: `${message} <br />Notebook saved`,
-          }))
-          dispatch({ type: 'ADD_NOTEBOOK_ID', id: json.id })
-          window.history.pushState('', {}, `/notebooks/${json.id}`)
-        })
+      createNewNotebookOnServer()(dispatch, getState)
     }
-    dispatch({ type: 'NOTEBOOK_SAVED' })
   }
 }
```

Comment:
This diff is awful.  I basically broke it up into two functions: create and save.